### PR TITLE
Perform schema composition at the end of the lifecycle

### PR DESCRIPTION
**Pull request by darrellwarde**
# Description

> **Note**
> 
> Please provide a description of the work completed in this PR below

Performs all schema composition (resolvers composition, adding of default resolvers) at the end of the schema generation lifecycle.

For Federation, this means that resolvers composition can be performed in one step, making that schema generation more aligned with standard schema generation.

# Complexity

Low

### CODE REVIEW

**Comment by neo4j-team-graphql** [comment]
Made some changes to the Neo4j GraphQL Toolbox? Or made some changes to the core library perhaps?

Check out your changes at https://neo4j-graphql-toolbox-3454.surge.sh

Use for instance a local dbms hosted on [Neo4j Desktop](https://neo4j.com/download/) or a cloud-hosted [Aura Free](https://neo4j.com/cloud/aura-free/) dbms.

**Comment by neo4j-team-graphql** [comment]
# Performance Report

No Performance Changes


<details>
    <summary>Show Full Table</summary>
    
| name | dbHits | old dbHits | time (ms) | old time (ms) | maxRows |
| ---- | ------ | ---------- | --------- | ------------- | ------- |
|  aggregations.TopLevelAggregate | 3403 | 3403 | 44 | 54 | 1134 |
|  aggregations.NestedAggregation | 16553 | 16553 | 89 | 104 | 2174 |
|  aggregations.AggregationWithWhere | 11979 | 11979 | 42 | 64 | 2174 |
|  aggregations.AggregationWhereWithinNestedRelationships | 22116987 | 22116987 | 3231 | 3362 | 2008534 |
|  aggregations.AggregationWhereWithinNestedConnections | 22116987 | 22116987 | 2898 | 2996 | 2008534 |
|  aggregations.NestedCountFromMovieToActors | 9734 | 9734 | 38 | 47 | 2174 |
|  aggregations.NestedCountFromActorsToMovie | 9937 | 9937 | 43 | 62 | 2174 |
|  aggregations.DeeplyNestedCount | 13070329 | 13070333 | 4526 | 4825 | 2008534 |
|  batch-create.BatchCreate | 4200 | 4200 | 132 | 148 | 600 |
|  batch-create.BatchCreateSmall | 77 | 77 | 53 | 108 | 11 |
|  connect.createAndConnect | 14424 | 14424 | 194 | 249 | 3003 |
|  connections.Connection | 14082 | 14082 | 58 | 70 | 2174 |
|  connections.NestedConnection | 45472 | 45472 | 103 | 131 | 4516 |
|  connections.ConnectionWithSort | 3284 | 3284 | 60 | 67 | 1040 |
|  connections.ConnectionWithSortAndCypher | 3284 | 3284 | 198 | 327 | 1040 |
|  create.SimpleMutation | 7 | 7 | 53 | 60 | 1 |
|  cypher-directive.TopLevelCypherDirective | 122100 | 122100 | 572 | 665 | 12241 |
|  cypher-directive.TopLevelCypherDirectiveWithColumnName | 153617 | 153617 | 174 | 183 | 12241 |
|  delete.SimpleDelete | 19401 | 19401 | 218 | 358 | 1040 |
|  delete.NestedDeleteInUpdate | 23202 | 23202 | 135 | 150 | 2040 |
|  2925.SingleRelationshipFilter | 6468 | 6468 | 65 | 46 | 1040 |
|  2925.NestedSingleRelationshipFilter | 22900 | 22900 | 89 | 111 | 2174 |
|  2925.SingleRelationshipRequiredFilter | 5201 | 5201 | 38 | 49 | 1040 |
|  2925.NestedSingleRelationshipRequiredFilter | 9361 | 9361 | 57 | 80 | 1040 |
|  query.SimpleQuery | 3121 | 3121 | 21 | 28 | 1040 |
|  query.SimpleQueryWithRelationship | 16162 | 16162 | 49 | 53 | 2174 |
|  query.QueryWhere | 9695 | 9719 | 35 | 51 | 2161 |
|  query.SimpleQueryWithNestedWhere | 9873 | 9897 | 51 | 68 | 2161 |
|  query.Nested | 10096041 | 10096041 | 10056 | 10165 | 2008534 |
|  query.NestedWithFilter | 10074401 | 10074401 | 10506 | 10139 | 2004000 |
|  query.OrFilterOnRelationships | 43666 | 43454 | 190 | 213 | 2042 |
|  query.OrFilterOnRelationshipsAndNested | 36915 | 37586 | 203 | 283 | 2042 |
|  query.QueryWithNestedIn | 13388 | 13363 | 53 | 66 | 1213 |
|  query.NestedConnectionWhere | 9834 | 9834 | 54 | 87 | 2174 |
|  query.DeeplyNestedConnectionWhere | 10019 | 10004 | 76 | 120 | 2174 |
|  query.DeeplyNestedWithRelationshipFilters | 19338 | 19212 | 140 | 204 | 1664 |
|  query.NestedWithRelationshipSingleFilters | 3881 | 3881 | 171 | 246 | 1134 |
|  query.Fulltext | 80 | 80 | 32 | 39 | 16 |
|  query.FulltextWithNestedQuery | 587 | 587 | 43 | 49 | 84 |
|  sorting.SortMultipleTypes | 3515 | 3515 | 64 | 77 | 1040 |
|  sorting.SortMultipleTypesWithCypherWithCypher | 1423 | 1425 | 213 | 337 | 1040 |
|  sorting.SortOnNestedFields | 14082 | 14082 | 56 | 62 | 2174 |
|  sorting.SortDeeplyNestedFields | 43198 | 43198 | 98 | 126 | 4516 |
|  sorting.SortWithTopLevelCypher | 3121 | 3121 | 130 | 207 | 1040 |
|  unions.SimpleUnionQuery | 321 | 321 | 60 | 69 | 35 |
|  unions.SimpleUnionQueryWithMissingFields | 293 | 293 | 58 | 62 | 35 |
|  unions.NestedUnion | 410637 | 410637 | 291 | 324 | 33033 |
|  unions.NestedUnionWithMissingFields | 384611 | 384611 | 269 | 303 | 33033 |
|  update.NestedUpdate | 16143 | 16143 | 95 | 113 | 2002 |

</details>
        
Old Schema Generation: 43.797s
Schema Generation: 44.860s
Old Subgraph Schema Generation: 56.468s
Subgraph Schema Generation: 57.157s

File: packages/graphql/src/classes/Neo4jGraphQL.ts [hunk 1]
```diff
@@ -307,28 +307,21 @@ class Neo4jGraphQL {
         return composeResolvers(mergedResolvers, resolversComposition);
     }
 
-    private wrapFederationResolvers(resolvers: NonNullable<IExecutableSchemaDefinition["resolvers"]>) {
-        if (!this.schemaModel) {
-            throw new Error("Schema Model is not defined");
-        }
+    private composeSchema(schema: GraphQLSchema): GraphQLSchema {
+        // if (this.middlewares) {
```
Comment (mjfwebb) [inline packages/graphql/src/classes/Neo4jGraphQL.ts:311]: Just for posterity, it might be nice to add a TODO here.

**Review by mjfwebb — APPROVED** [review]
Great work!

**Pull request merged**